### PR TITLE
Updated Filing Error /  Payment Error Handling

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#13838

*Description of changes:*
* Implements error handling on MHR filings for save and payment errors.
* Updated Generic error handling and message templating.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/App.vue [hunk 1]
```diff
@@ -589,6 +594,7 @@ export default defineComponent({
       if (localState.registrationTypeUI) {
         localState.payErrorOptions.text = localState.payErrorOptions.text.replace('filing_type', filing)
       }
+      console.log(error)
```
Comment (dimak1) [inline ppr-ui/src/App.vue]: On purpose or needs to be cleaned up?

File: ppr-ui/src/components/common/ButtonFooter.vue [hunk 2]
```diff
@@ -135,24 +135,22 @@ export default defineComponent({
...
 
     const {
-      resetNewRegistration, setDraft, setRegTableNewItem, setUnsavedChanges, setMhrDraftNumber
+      resetNewRegistration, setDraft, setRegTableNewItem, setUnsavedChanges
     } = useActions<any>([
-      'resetNewRegistration', 'setDraft', 'setRegTableNewItem', 'setUnsavedChanges', 'setMhrDraftNumber'
+      'resetNewRegistration', 'setDraft', 'setRegTableNewItem', 'setUnsavedChanges'
     ])
-    const { buildApiData } = useNewMhrRegistration()
+    const { mhrDraftHandler } = useNewMhrRegistration()
```
Comment (dimak1) [inline ppr-ui/src/components/common/ButtonFooter.vue:153]: 👍🏻

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1296-gup7ib7h.web.app

**Review by dimak1 — APPROVED** [review]
Looks good!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1296-gup7ib7h.web.app

File: ppr-ui/src/App.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/App.vue]: Cleaned up!

**Review by doug-lovett — APPROVED** [review]
Looks great.

**Pull request merged**